llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -7067,30 +7067,53 @@ class BoUpSLP::ShuffleCostEstimator : public BaseShuffleAnalysis {
   /// extracted values from \p VL.
   InstructionCost computeExtractCost(ArrayRef<Value *> VL, ArrayRef<int> Mask,
                                      TTI::ShuffleKind ShuffleKind) {
-    auto *VecTy = FixedVectorType::get(VL.front()->getType(), VL.size());
+    unsigned NumElts = 0;
+    for (Value *V : VL) {
+      auto *EE = dyn_cast<ExtractElementInst>(V);
+      if (!EE)
+        continue;
+      auto *VecTy = cast<FixedVectorType>(EE->getVectorOperandType());
+      NumElts = std::max(NumElts, VecTy->getNumElements());
+    }
+    auto *VecTy = FixedVectorType::get(VL.front()->getType(), NumElts);
     unsigned NumOfParts = TTI.getNumberOfParts(VecTy);
-
-    if (ShuffleKind != TargetTransformInfo::SK_PermuteSingleSrc ||
-        !NumOfParts || VecTy->getNumElements() < NumOfParts)
+    if (!NumOfParts || NumElts < NumOfParts)
```

Comment:
When the condition NumElts < NumOfParts is true (assuming NumElts > 0 and NumOfParts is non-zero)?

Reply:
Found the condition originally appeared with 
```
commit 8e5f3d04f269dbe791076e775f1d1a098cbada01
Author: Fangrui Song <i@maskray.me>
Date:   Fri Apr 2 11:13:51 2021 -0700

    [SLPVectorizer] Fix divide-by-zero after D99719

    Will add a test case later.
```
But it looks like there is still no test case that executes the condition. @MaskRay , any chance to find/add the test case?